common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -5536,10 +5536,12 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
 
     HIVE_QUERY_REEXECUTION_ENABLED("hive.query.reexecution.enabled", true,
         "Enable query reexecutions"),
-    HIVE_QUERY_REEXECUTION_STRATEGIES("hive.query.reexecution.strategies", "overlay,reoptimize,reexecute_lost_am,dagsubmit",
+    HIVE_QUERY_REEXECUTION_STRATEGIES("hive.query.reexecution.strategies",
+        "overlay,reoptimize,reexecute_lost_am,dagsubmit,recompile_without_cbo",
         "comma separated list of plugin can be used:\n"
             + "  overlay: hiveconf subtree 'reexec.overlay' is used as an overlay in case of an execution errors out\n"
             + "  reoptimize: collects operator statistics during execution and recompile the query after a failure\n"
+            + "  recompile_without_cbo: recompiles query after a CBO failure\n"
             + "  reexecute_lost_am: reexecutes query if it failed due to tez am node gets decommissioned"),
```

Comment:
I didn't go through the whole PR but I get the impression that we could/should combine the `hive.query.reexecution.strategies` and `hive.cbo.fallback.strategy` configurations somehow. Having both does not seem necessary and raises some questions about the expected behavior.

Consider for instance the following:
```
set hive.cbo.fallback.strategy = always;
-- Note that recompile_without_cbo is missing 
set hive.query.reexecution.strategies = overlay,reoptimize,reexecute_lost_am,dagsubmit;
```
Reading the current configuration I am not sure what should happen when the CBO fails.

The `hive.cbo.fallback.strategy` has not been released yet so we are free to drop it, or modify it to be consistent with `hive.query.reexecution.strategies`.

Reply:
That's a good point. I think we should check the 2 configurations, and fail if they are not aligned.
What do you think?

Reply:
Added tests